### ISSUE TITLE
Spring Data REST force using of jackson serialization

### ISSUE THREAD

**Issue by xanscale** [spring-projects/spring-data-rest#2439]
there is no way to change json serialization, like kotlinx-serialization or gson or others.

inside `RepositoryRestMvcConfiguration` jackson seams hardcoded and no way to change.

https://docs.spring.io/spring-framework/reference/languages/kotlin/web.html#kotlin-multiplatform-serialization

**Comment by mp911de** [comment]
Exactly. Spring Data REST is built on top of Jackson, and we use Jackson in various areas. It would require a significant rework to support other serialization mechanisms.

**Comment by xanscale** [comment]
It's a very unexpected answer :) 
in my mind data REST is on top of RestController, something like autogenerated standard RestController

at least it should be specified in the documentation

I'm making rest api with data REST and kotlin, but Jackson and Kotlin are not good friends
